frontend/src/components/editor/cell/code/cell-editor.tsx

```diff
@@ -335,17 +336,26 @@ const CellEditorInternal = ({
     };
   }, [editorViewRef]);
 
-  const temporarilyShowCode = async () => {
+  const temporarilyShowCode = useCallback(async () => {
     if (hidden) {
       updateCellConfig({ cellId, config: { hide_code: false } });
       editorViewRef.current?.focus();
       editorViewParentRef.current?.addEventListener(
-        "blur",
+        "focusout",
         () => updateCellConfig({ cellId, config: { hide_code: true } }),
         { once: true },
       );
     }
-  };
+  }, [hidden, cellId, updateCellConfig, editorViewRef]);
+
+  // For a newly created Markdown cell, which defaults to
+  // hidden code, we temporarily show the code editor
+  useEffect(() => {
+    if (!(isMarkdown(code) && code === `mo.md(r"""\n""")`)) {
+      return;
+    }
+    temporarilyShowCode();
+  }, [code, temporarilyShowCode]);
```

Comment:
i might be better to just remove `temporarilyShowCode` and add the eslint disable. clearer that it only react to code changes